http/rest-client-reactive/src/test/java/io/quarkus/ts/http/restclient/reactive/UrlOverrideClientIT.java

```diff
@@ -0,0 +1,127 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.JaegerService;
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.JaegerContainer;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+@Tag("QUARKUS-5612")
+public class UrlOverrideClientIT {
+    private static final String MAIN_SERVICE_NAME = "overridingService";
+
+    // used for jaeger filtering, to only query traces relevant for current test
+    private long testStartTime = 0;
+
+    /*
+     * Start two application. By default, restClient in app will do HTTP requests against app.
+     * But if we override the URL, it will do requests against app2.
+     * Both apps will be distinguished by value of "ts.quarkus.urlOverride.response" which they return as HTTP response.
+     * The URL override feature overrides base URL, but not HTTP path. So we cannot easily test it using just one app.
+     */
+    @JaegerContainer(expectedLog = "\"Health Check state change\",\"status\":\"ready\"")
+    static final JaegerService jaeger = new JaegerService();
+
+    @QuarkusApplication()
+    static RestService app = new RestService()
+            .withProperties("urlOverride.properties")
+            .withProperty("quarkus.otel.enabled", "true")
+            .withProperty("quarkus.otel.simple", "true")
+            .withProperty("quarkus.application.name", MAIN_SERVICE_NAME)
+            .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", jaeger::getCollectorUrl);
+
+    @QuarkusApplication()
+    static RestService app2 = new RestService()
+            .withProperties("urlOverride.properties")
+            .withProperty("quarkus.http.ssl-port", "8444")
```

Comment:
I don't understand why are you configuring HTTPS port but never use it.

Reply:
App requires it. I'm not entirely sure where it is coming from, but at the start both apps try co claim port 8443 (if not changed) and one fails.

Reply:
You don't use HTTPS in your tests, if it comes from some global module application.properties, you could have disabled it or add comment because it feels confusing. Thanks for the explanation though, I think it is OK.